src/Integration/Notifications/VisualStudioStatusBarHelper.cs

```diff
@@ -0,0 +1,88 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Media;
+
+namespace SonarLint.VisualStudio.Integration.Notifications
+{
+    public static class VisualStudioStatusBarHelper
+    {
+        private const string SccStatusBarHostName = "PART_SccStatusBarHost";
+        private const string StatusBarPanelName = "StatusBarPanel";
+
+        public static void RemoveStatusBarIcon(FrameworkElement statusBarIcon)
+        {
+            var dockPanel = statusBarIcon.Parent as DockPanel;
+            if (dockPanel == null)
+            {
+                return;
+            }
+
+            dockPanel.Children.Remove(statusBarIcon);
+        }
+
+        public static void AddStatusBarIcon(FrameworkElement statusBarIcon)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Consider using more general type 'System.Windows.UIElement' instead of 'System.Windows.FrameworkElement'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS3242)